consensus/src/experimental/tests/commit_phase_tests.rs

```diff
@@ -0,0 +1,235 @@
+        Waypoint::new_epoch_boundary(&LedgerInfo::mock_genesis(Some(validator_set))).unwrap();
+
+    let safety_storage = PersistentSafetyStorage::initialize(
+        Storage::from(diem_secure_storage::InMemoryStorage::new()),
+        signer.author(),
+        signer.private_key().clone(),
+        Ed25519PrivateKey::generate_for_testing(),
+        waypoint,
+        true,
+    );
+    let safety_rules_manager =
+        SafetyRulesManager::new_local(safety_storage, false, false, true);
+
+    let (initial_data, storage) = MockStorage::start_for_testing((&validators).into());
+    let epoch_state = EpochState {
+        epoch: 1,
+        verifier: storage.get_validator_set().into(),
+    };
+    let validators = epoch_state.verifier.clone();
+    let (network_reqs_tx, network_reqs_rx) = diem_channel::new(QueueStyle::FIFO, 8, None);
+    let (connection_reqs_tx, _) = diem_channel::new(QueueStyle::FIFO, 8, None);
+
+    let network_sender = ConsensusNetworkSender::new(
+        PeerManagerRequestSender::new(network_reqs_tx),
+        ConnectionRequestSender::new(connection_reqs_tx),
+    );
+    let author = signer.author();
+
+    let (self_sender, self_receiver) = channel::new_test(1000);
+    let network = NetworkSender::new(author, network_sender, self_sender, validators);
+
+    let (commit_cb_sender, commit_cb_receiver) = mpsc::unbounded::<LedgerInfoWithSignatures>();
+    let (state_sync_client, state_sync_receiver) = mpsc::unbounded();
+    let state_computer = Arc::new(MockStateComputer::new(
+        state_sync_client,
+        commit_cb_sender,
+        Arc::clone(&storage),
+    ));
+
+    let mut safety_rules =
+        MetricsSafetyRules::new(safety_rules_manager.client(), storage.clone());
+    safety_rules.perform_initialize().unwrap();
+
+    let safety_rules_container = Arc::new(Mutex::new(safety_rules));
+
+    let time_service = Arc::new(ClockTimeService::new(executor));
+
+    let block_store = Arc::new(BlockStore::new(
+        storage.clone(),
+        initial_data,
+        state_computer.clone(),
+        10, // max pruned blocks in mem
+        time_service.clone(),
+    ));
+
+    let proposal_generator = ProposalGenerator::new(
+        author,
+        block_store.clone(),
+        Arc::new(MockTransactionManager::new(None)),
+        time_service.clone(),
+        1,
+    );
+
+    // Setting up channels
+
+    let guage_c = IntGauge::new(
+        "D_COM_CHANNEL_COUNTER_EM",
+        "counter for the decoupling committing channel in epoch manager",
+    ).unwrap();
+
+    let (sender_comm, receiver_comm) = channel::new::<(
+        Vec<ExecutedBlock>,
+        LedgerInfoWithSignatures,
+    )>(channel_size, &guage_c);
+
+    let guage_c_msg = IntGauge::new(
+        "D_COM_CHANNEL_COUNTER_EM",
+        "counter for the decoupling committing channel in epoch manager",
+    )
+        .unwrap();
+    let (sender_c_msg, receiver_c_msg) =
+        channel::new::<ConsensusMsg>(channel_size, &guage_c_msg);
```

Comment:
there's some utilities function of creating test channels

Reply:
resolved